prow/cmd/deck/tide.go

```diff
@@ -147,68 +151,118 @@ func (ta *tideAgent) updateHistory() error {
 	if err := fetchTideData(ta.log, path, &history); err != nil {
 		return err
 	}
-	history = ta.filterHiddenHistory(history)
+	history = ta.filterHistory(history)
 
 	ta.Lock()
 	defer ta.Unlock()
 	ta.history = history
 	return nil
 }
 
-func (ta *tideAgent) filterHiddenPools(pools []tide.Pool) []tide.Pool {
-	if len(ta.hiddenRepos()) == 0 {
-		return pools
-	}
+func (ta *tideAgent) matchingIDs(ids []string) bool {
+	return len(ids) > 0 && sets.String{}.Insert(ta.tenantIDs...).HasAll(ids...)
```

Comment:
nit: `sets.String{}.Insert(...)` can be simplified to `sets.NewString(...)`

It would be simpler and more efficient to just make `ta.tenantIDs` a `sets.String`.